playwright/src/test/java/com/microsoft/playwright/TestPageClock.java

```diff
@@ -290,8 +290,9 @@ void shouldRunTimeBeforePopup(Page page, Server server) {
     Page popup = page.waitForPopup(() -> {
       page.evaluate("url => window.open(url)", server.PREFIX + "/popup.html");
     });
+    popup.waitForURL(server.PREFIX + "/popup.html");
```

Comment:
Why don’t we need it upstream? https://github.com/microsoft/playwright/blob/77dee44984af1cd5b7c1ec4f6ed6fe022c7c55e6/tests/library/page-clock.spec.ts#L339

Reply:
We do.